### PR TITLE
[Task #1855259, #1905702] Implement lazy loading in app service combo box

### PR DESCRIPTION
**Pull request by Flanker32**
- Support lazy loading in web app combo box
- Migrate to `AppServiceConfig` in `AppServiceComboBox` and deprecate `AppServiceComboBoxModel`

[AB#1855259](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1855259), [AB#1905702](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1905702)

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/resources/com/microsoft/intellij/ui/messages/messages.properties [hunk 1]
```diff
@@ -1481,9 +1484,11 @@ webapp.deploy.validate.noResourceGroup=Resource Group not provided.
 webapp.deploy.validate.noAppServicePlan=App Service Plan not provided.
 webapp.deploy.validate.noLocation=Location not provided.
 webapp.deploy.validate.noPricingTier=Pricing Tier not provided.
+webapp.deploy.validate.loading=Loading meta-data of target web app
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/resources/com/microsoft/intellij/ui/messages/messages.properties]: `webapp.validation_runconfiguration.loading`, e.g. meta-data of target webapp is still loading...`

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/intellij/legacy/appservice/AppServiceComboBox.java [hunk 2]
```diff
@@ -9,107 +9,109 @@
...
                 items.add(configModel);
             }
         }
         return items;
     }
 
+    @Override
+    public T getValue() {
+        if (value instanceof ItemReference && ((ItemReference<?>) value).is(configModel)) {
+            return configModel;
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/intellij/legacy/appservice/AppServiceComboBox.java:81]: `((ItemReference<?>) value).is(configModel)` is also `true` if `value`(the `ItemReference`) refers to an existing app which has same name or id as the `configModel`. but it should return the existing app but not the `configModel`

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/intellij/legacy/appservice/AppServiceComboBox.java [hunk 3]
```diff
@@ -9,107 +9,109 @@
...
     protected abstract void createResource();
 
     public static class AppComboBoxRender extends SimpleListCellRenderer {
 
         @Override
-        public void customize(JList list, Object value, int index, boolean b, boolean b1) {
-            if (value instanceof AppServiceComboBoxModel) {
-                final AppServiceComboBoxModel app = (AppServiceComboBoxModel) value;
+        public void customize(JList list, Object value, int index, boolean isSelected, boolean cellHasFocus) {
+            if (value instanceof AppServiceConfig) {
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/intellij/legacy/appservice/AppServiceComboBox.java]: is it possible not a `AppServiceConfig`? and how should we render it if it's not an `AppServiceConfig`

Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/intellij/legacy/appservice/AppServiceComboBox.java]: it should always be AppServiceConfig but may be null here

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/intellij/legacy/function/FunctionAppComboBox.java [hunk 4]
```diff
@@ -41,11 +45,27 @@ protected void createResource() {
...
+                .subscription(Subscription.builder().id(functionApp.subscriptionId()).build()).build();
+        AzureTaskManager.getInstance()
+                .runOnPooledThreadAsObservable(new AzureTask<>(functionApp::entity))
+                .subscribe(entity -> {
+                    result.setRuntime(entity.getRuntime());
+                    result.setRegion(entity.getRegion());
+                    result.setServicePlan(AppServicePlanEntity.builder().id(entity.getAppServicePlanId()).build());
+                });
+        return result;
+    }
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/intellij/legacy/function/FunctionAppComboBox.java]: can this method be shared between webapp and function app?

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/intellij/legacy/appservice/AppServiceComboBox.java (hunk 2 above)
Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/intellij/legacy/appservice/AppServiceComboBox.java:81]: This is to return value during combobox is still refreshing, will clean up the configModel reference during `loadItems()`

**Review by wangmingliang-ms — APPROVED** [review]

**Pull request merged**